src/NGen.cpp

```diff
@@ -51,7 +52,9 @@ bool is_subdivided_hydrofabric_wanted = false;
 std::string PARTITION_PATH = "";
 int mpi_rank;
 int mpi_num_procs;
-#endif
+#else // NGEN_MPI_ACTIVE
+int mpi_rank = 0;
+#endif // NGEN_MPI_ACTIVE
 
```

Comment:
if mpi_rank is used regardless of the macro value, declare & define once outside the macro logic.
```suggestion
int mpi_rank = 0;
```

Reply:
Sure